### DISCUSSION TITLE
Support for polygon intersection query

### DISCUSSION THREAD

**Discussion by manorbar** [coady/lupyne#22] [Ideas]
My team is currently on the lookout for alternatives to spatial index creation using Python and lucene (and then lupyne) came up.
currently we use RTree to create in memory spatial index with between 50K-150K polygons and for each polygon we search the intersected polygons and run some logic (which is really inefficient and happens in one process due to RTree limitations). 
I saw that SpatialField has within (which could work with some sort of buffer on the geometry but we would rather avoid it if possible) and distance (which doesn't seem like a good fit for our needs) query.  
Wondering what do you think about adding support for Polygons and intersection query  
Thanks in advance

**Comment by coady** [comment]
Yeah, the original `SpatialField` predates all the shape fields that have since been added to lucene. But if there's interest I'm fine with adding support for the new fields.

**Reply by manorbar** [reply]
That would be amazing and a total game changer for us  
is it possible to contribute to the project or are you working alone?

**Reply by coady** [reply]
Contributions are welcome, but I'd like to add the initial `Field`. #23

**Reply by manorbar** [reply]
Looks amazing, thanks a lot!
